src/test/rpc_tests.cpp

```diff
@@ -280,11 +280,16 @@ BOOST_AUTO_TEST_CASE(rpc_ban)
     ar = r.get_array();
     o1 = ar[0].get_obj();
     adr = find_value(o1, "address");
-    banned_until = find_value(o1, "banned_until");
+    banned_until = find_value(o1, "banned_until").get_int64();
+    const int64_t ban_created{find_value(o1, "ban_created").get_int64()};
+    const int64_t ban_duration{find_value(o1, "ban_duration").get_int64()};
+    const int64_t time_remaining{find_value(o1, "time_remaining").get_int64()};
+    const int64_t now{GetTime()};
     BOOST_CHECK_EQUAL(adr.get_str(), "127.0.0.0/24");
-    int64_t now = GetTime();
-    BOOST_CHECK(banned_until.get_int64() > now);
-    BOOST_CHECK(banned_until.get_int64()-now <= 200);
+    BOOST_CHECK(banned_until > now);
+    BOOST_CHECK(banned_until - now <= 200);
+    BOOST_CHECK_EQUAL(ban_duration, banned_until - ban_created);
+    BOOST_CHECK_EQUAL(time_remaining, banned_until - now);
```

Comment:
the may fail:

```
Running 1 test case...
test/rpc_tests.cpp(293): error: in "rpc_tests/rpc_ban": check time_remaining == banned_until - now has failed [200 != 199]

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
```

diff:

```diff
diff --git a/src/test/rpc_tests.cpp b/src/test/rpc_tests.cpp
index 3b6faf7bbb..be0d79aa86 100644
--- a/src/test/rpc_tests.cpp
+++ b/src/test/rpc_tests.cpp
@@ -277,6 +277,7 @@ BOOST_AUTO_TEST_CASE(rpc_ban)
 
     BOOST_CHECK_NO_THROW(r = CallRPC(std::string("setban 127.0.0.0/24 add 200")));
     BOOST_CHECK_NO_THROW(r = CallRPC(std::string("listbanned")));
+    std::this_thread::sleep_for(std::chrono::milliseconds(1100)); // CPU scheduler puts this thread to sleep
     ar = r.get_array();
     o1 = ar[0].get_obj();
     adr = find_value(o1, "address");
```

Can be fixed by using mocktime (SetTime()), or by relaxing the test.

Also, the whole test can be moved to the functional test suite. For example, `./test/functional/rpc_setban.py`

Reply:
See #21676